include/openthread/nat64.h

```diff
@@ -88,6 +88,127 @@ typedef struct otIp4Cidr
     uint8_t      mLength;
 } otIp4Cidr;
 
+/**
+ * @struct otNat64Counters
+ *
+ * This structure represents the counters for NAT64.
+ *
+ */
+typedef struct otNat64Counters
+{
+    uint64_t m4To6SumPackets;
+    uint64_t m4To6SumBytes;
+    uint64_t m6To4SumPackets;
+    uint64_t m6To4SumBytes;
+} otNat64Counters;
+
+/**
+ * @struct otNat64ProtocolCounters
+ *
+ * This structure represents the counters for the protocols supported by NAT64.
+ *
+ */
+typedef struct otNat64ProtocolCounters
+{
+    otNat64Counters mTotal;
+    otNat64Counters mIcmp;
+    otNat64Counters mUdp;
+    otNat64Counters mTcp;
+} otNat64ProtocolCounters;
+
+/**
+ * Packet drop reasons.
+ *
+ */
+typedef enum otNat64DropReason
+{
+    OT_NAT64_DROP_REASON_UNKNOWN = 0,       ///< Packet drop for unknown reasons.
+    OT_NAT64_DROP_REASON_ILLEGAL_PACKET,    ///< Packet drop due to failed to parse the datagram.
+    OT_NAT64_DROP_REASON_UNSUPPORTED_PROTO, ///< Packet drop due to unsupported IP protocol.
+    OT_NAT64_DROP_REASON_NO_MAPPING, ///< Packet drop due to no corresponding mappings found, or address mapping pool
```

Comment:
Would be great if we can fit this in one line. Maybe we can just explain the readon and remove "Packet drop due" from all.
For example, we can say "No mappings found or mapping pool exhuasted"